app/code/Magento/QuoteGraphQl/Model/Cart/QuoteAddressFactory.php

```diff
@@ -61,7 +61,11 @@ public function __construct(
     public function createBasedOnInputData(array $addressInput): QuoteAddress
     {
         $addressInput['country_id'] = '';
-        if ($addressInput['country_code']) {
+        if (!isset($addressInput['country_code'])) {
```

Comment:
Please leave only `isset` check on line 64. `country_id` is kept here for BC purposes.
Here is the related task https://github.com/magento/graphql-ce/issues/914